### PR TITLE
Embed generated openapi.json in Go binary

### PR DESCRIPTION
**Pull request by djjuhasz**
Embed `openapi.json` in the Go binary so it can be served via the
Enduro API as `/swagger/swagger.json`.

### CODE REVIEW

**Comment by djjuhasz** [comment]
@sevein I tried embedding the OpenAPI files generated by Goa, and then serving the "openapi.json" file from the "/swagger/swagger.json" URL, but I get a "404 page not found" response. :( 

I can't see what I'm doing wrong. Could you please have a look?

File: internal/api/api.go [hunk 1]
```diff
@@ -33,6 +34,9 @@ import (
 	intupload "github.com/artefactual-sdps/enduro/internal/upload"
 )
 
+// go:embed gen/http/*.*
```
Comment (sevein) [inline internal/api/api.go]: I think that the main issue is in that extra space after `//`. It should read:

```go
//go:embed gen/http/*.*
```

But you probably want something like this to embed just that file:

```go
//go:embed gen/http/openapi.json
```

File: internal/api/design/swagger.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ var _ = Service("swagger", func() {
 	HTTP(func() {
 		Path("/swagger")
 	})
-	Files("/swagger.json", "internal/api/gen/http/openapi.json", func() {
+	Files("/swagger.json", "openapi.json", func() {
```
Comment (sevein) [inline internal/api/design/swagger.go]: I think that the second parameter should read: `gen/http/openapi.json`.

That would be the path used to access the embedded filesystem.

File: internal/api/api.go (hunk 1 above)
Comment (djjuhasz) [reply internal/api/api.go]: Done. :)

File: internal/api/design/swagger.go (hunk 2 above)
Comment (djjuhasz) [reply internal/api/design/swagger.go]: Ah! That did the trick, thanks!! :relieved:

**Review by sevein — APPROVED** [review]

**Pull request merged**